pkg/component/runtime/manager_test.go

```diff
@@ -3304,7 +3304,7 @@ LOOP:
 	require.NoError(t, err)
 }
 
-func TestManager_FakeInput_OutputChange(t *testing.T) {
+func TestManager_FakeInput_StopInput0StartInput1(t *testing.T) {
```

Comment:
```suggestion
// Verifies that components stopped before they are started.
func TestManager_FakeInput_StopInput0StartInput1(t *testing.T) {
```

This test needs a description comment as well since it is so complicated. The test was originally added in https://github.com/elastic/elastic-agent/pull/2226 to verify that the component start and stop order is correct.

Reply:
Unfortunately (unless something changed since I chatted with Lee?) this ordering constraint is probably what caused the race condition. The first component is stopped before the second is started, but there's no way to verify the ordering by listening to state changes, because the two components are managed by different goroutines, and so the "stopped" and "started" states can be sent in parallel even if there is a deterministic order to the actions that caused them. In the current code the state subscriptions only reliably represent the ordering of states within a single component, not between distinct components.